docs/Bridging_Assets.md

```diff
@@ -110,3 +110,29 @@ Note that in the system described above, one pair of Gateway contracts handles t
 Take our wrapped Ether implementation, for example: here, a single WETH contract on L1 is connected to a single WETH contract on L2. When transferring WETH from one domain to another, the L1/L2 Gateway architecture is used to unwrap the WETH on domain A, transfer the now-unwrapped Ether, and then re-wrap it on domain B. This ensures that WETH can behave on Arbitrum the way users are used to it behaving on Ethereum, while ensuring that all WETH tokens are always fully collateralized on the layer in which they reside.
 
 No matter the complexity of a particular token's bridging needs, a gateway can in principle be created it to accommodate it within our canonical bridging system.
+
+### The Arbitrum Generic Custom Gateway
+
+Just because a token has requirements beyond what are offered via the StandardERC20 gateway, that doesn't necessarily mean that a unique Gateway needs to be taylor-made for the token in question. Our Generic-Custom Gateway is designed to be flexible enough to be suitable for most (but not necessarily all) custom fungible token needs. As a general rule:
+
+**If your token's total supply on L2 only ever increases when tokens are deposited from L1 (i.e., no L2-interest accrual, no special L2 minting affordance, etc), but your token requires customization beyond what the StandardERC20 Gateway provides, the Generic-Custom Gateway is likely the right solution for you!**
+
+Some examples of token features suitable for the Generic-Custom Gateway:
+
+- L2 token contract upgradable via a proxy
```

Comment:
We could add that you can use custom tokens if you want to determine the L2 token address yourself

Reply:
ah yeah, i like that

Reply:
That's a better 3rd option for the list; the current one is week, but just wanted 3 for aesthetic rule-of-3 reasons